src/main/java/com/avioconsulting/mule/logger/internal/listeners/CustomLoggerPipelineNotificationListener.java

```diff
@@ -49,11 +68,17 @@ public void onNotification(PipelineMessageNotification notification) {
             return;
         }
         classLogger.debug(message);
+        Map<String, String> flowLogAttributes = getFlowLogAttributes(notification);
         logMessage(notification.getComponent().getLocation(), notification.getEvent(), message,
             config.getFlowCategorySuffix(),
-            config.getFlowLogLevel());
+            config.getFlowLogLevel(), flowLogAttributes);
       } catch (Exception e) {
-        classLogger.error("Error processing flow notification", e);
+        if (e.getClass().getName().equals("java.lang.ClassCastException")
+            || e.getClass().getName().equals("ClassCastException")) {
+          classLogger.error("Message expression text in flow-log-config needs to be a String", e);
```

Comment:
Can we add a separate `catch(ClassCastException castEx)` block to handle this error?

Reply:
existing catch block handle general Exception which covers the ClassCastException, so using the if condition in the same catch block

Reply:
You can retain the existing and add the specific ones before the general exception, those are handled in ordered specific to general -
```
try {

} catch (ClassCastException castEx) {
// Handle class cast only
} catch (Exception ex){
// existing general exception
}
```

Reply:
thanks and updated